File: Swift/factors.swift
import Foundation

let start = Date().timeIntervalSince1970


let num = 2000000000

for i in 1...num+1 {
    if num % i == 0 {
        print("\(i) is a factor of \(num)")
    }
}

let time = NSDate().timeIntervalSince1970 - start

let time_seconds = time

print("Completed in: \(time_seconds)s")

File: OCaml/factors.ml
let get_current_timestamp () : float =
  Unix.gettimeofday () ;;

let num = 2000000000 ;;

let rec get_factors idx = 
  if idx > num then 
    ()
  else (
    if num mod idx = 0 then (
      Printf.printf "%i is a factor of %i\n" idx num;
      flush stdout;
    );
    get_factors (idx + 1)
    ) ;;


let () = 
  let start = get_current_timestamp () in
  get_factors 1;
  let elapsed = get_current_timestamp () -. start in
  Printf.printf "Completed in: %.3fs" elapsed ;;
